qbraid/transpiler/conversions/openqasm3/openqasm3_to_cudaq.py

```diff
@@ -0,0 +1,172 @@
+                raise ProgramConversionError(
+                    f"Multi-dimensional array indexing is unsupported: {qubit.indices}"
+                )
+
+            inds = qubit.indices[0]
+            if isinstance(inds, ast.DiscreteSet):
+                inds = inds.values
+
+            if len(inds) > 1:
+                raise ProgramConversionError(
+                    f"Discrete set or multi-integer indexing is unsupported: {inds}"
+                )
+
+            ind = inds[0]
+            if not isinstance(ind, ast.IntegerLiteral):
+                raise ProgramConversionError(f"Index must be a single integer literal: {ind}")
+
+            q = ctx[qubit.name.name][ind.value]
+        else:
+            assert isinstance(qubit, ast.Identifier)
+            q = ctx[qubit.name][0]
+        return q
+
+    for statement in program.statements:
+        if isinstance(statement, ast.Include):
+            if statement.filename == "stdgates.inc":
+                continue
+            raise ProgramConversionError(f"Includes are unsupported: {statement}")
+        if isinstance(statement, ast.QubitDeclaration):
+            ctx[statement.qubit.name] = kernel.qalloc(statement.size.value)
+        elif isinstance(statement, ast.ClassicalDeclaration):
+            if not isinstance(statement.type, ast.ClassicalType):
+                raise ProgramConversionError(f"Unsupported statement: {statement}")
+
+            if statement.init_expression and isinstance(
+                statement.init_expression, ast.QuantumMeasurement
+            ):
+                ctx[statement.identifier.name] = kernel.mz(
+                    qubit_lookup(statement.init_expression.qubit)
+                )
+            else:
+                ctx[statement.identifier.name] = None
+        elif isinstance(statement, ast.QuantumMeasurementStatement):
+            val = kernel.mz(qubit_lookup(statement.measure.qubit))
+            if statement.target is not None:
+                ctx[identifier_lookup(statement.target)] = val
+        elif isinstance(statement, ast.QuantumGate):
+            name, qubits = statement.name.name, statement.qubits
+
+            # TODO: handle binary operations and special constants: e.g. pi/4
+            args = []
+            for arg in statement.arguments:
+                if arg.value is None:
+                    raise ProgramConversionError(
+                        f"Non-literal gate arguments are unsupported. {statement.arguments}"
+                    )
+                args.append(arg.value)
+
+            qubit_refs = [qubit_lookup(q) for q in qubits]
+
+
+            if len(statement.modifiers) > 1:
+                raise ProgramConversionError(
+                    f"Multiple gate modifiers are not supported: {statement}"
+                )
+            elif len(statement.modifiers) == 1:
+                mod = statement.modifiers[0]
+                if mod.modifier != ast.GateModifierName.ctrl:
```

Comment:
This can be extended to use [the logic in `pyqasm`](https://github.com/qBraid/pyqasm/blob/main/src/pyqasm/visitor.py#L922) unroller. Essentially, we evaluate any `power` modifiers and see if we have an odd number of `inv` modifiers. I presume that if we're unrolling first then this would not be an issue as we won't propagate these modifiers here